packages/test/test-end-to-end-tests/src/test/entryPointCompat.spec.ts

```diff
@@ -0,0 +1,118 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { describeLoaderCompat, describeNoCompat } from "@fluid-internal/test-version-utils";
+import {
+	ContainerRuntimeFactoryWithDefaultDataStore,
+	DataObject,
+	DataObjectFactory,
+} from "@fluidframework/aqueduct";
+import { IContainer } from "@fluidframework/container-definitions";
+import { IContainerRuntime } from "@fluidframework/container-runtime-definitions";
+import { FluidObject } from "@fluidframework/core-interfaces";
+import { ITestObjectProvider } from "@fluidframework/test-utils";
+
+describe("entryPoint compat", () => {
+	let provider: ITestObjectProvider;
+
+	class TestDataObject extends DataObject {
+		public get _root() {
+			return this.root;
+		}
+		public get _context() {
+			return this.context;
+		}
+	}
+
+	async function getDefaultFluidObject(runtime: IContainerRuntime): Promise<FluidObject> {
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		return (await runtime.getAliasedDataStoreEntryPoint?.("default"))!.get();
```

Comment:
It's awkward but we should probably follow up in next to remove the `?` since it's mandatory there now.